.github/actions/aderyn/action.yaml

```
@@ -0,0 +1,43 @@
+name: "Aderyn Static Analysis"
+description: "Run Aderyn static analysis and upload the report as an artifact"
+
+inputs:
+  rust_toolchain:
+    description: "Rust toolchain version"
+    required: true
+    default: "stable"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v4
+
+    - name: Setup Rust
+      uses: dtolnay/rust-toolchain@stable
+      with:
+        toolchain: ${{ inputs.rust_toolchain }}
+
+    - name: Install Aderyn
+      run: cargo install --git https://github.com/Cyfrin/aderyn.git
+      shell: bash
+
+    - name: Generate Aderyn Report
+      id: run_aderyn
+      run: |
+        aderyn --output report.md . 2>&1 | tee aderyn_output.log
+      shell: bash
+
+    - name: Upload Aderyn Report as Artifact
+      uses: actions/upload-artifact@v4
+      with:
+        name: aderyn-report
+        path: report.md
+        
+    - name: Check Aderyn Report for Warnings
+      run: |
+        if grep -q "warning:" aderyn_output.log || grep -q "Found issues" aderyn_output.log; then
+          echo "Aderyn analysis found warnings."
+          exit 1
+        fi
```

Maybe you can add `else` statement if there are no warnings or issues
```suggestion
        else
          echo "No warnings or issues found by Aderyn."
        fi
```